src/test/java/com/dpgil/pathlinker/path_linker/internal/PathLinkerModelTest.java

```diff
@@ -0,0 +1,594 @@
+		originalNetworkMap = new HashMap<String, Long>();
+		originalNetworkUndir = support.getNetwork();
+		originalNetworkUndir.getDefaultEdgeTable().createColumn("edge_weight", Double.class, false);
+
+		//access input file
+		url = PathLinkerModelTest.class.getResource("/input/graph-undir_human-interactome.txt");
+		file = new File(url.getFile());
+
+		//construct network based on input file
+		readerInput = new BufferedReader(new FileReader(file));
+
+		for (String line = readerInput.readLine(); line != null; line = readerInput.readLine()) {
+			String[] arr = line.split("\\s+");
+			CyNode source, target;
+
+			//to avoid adding same source
+			if (!originalNetworkMap.containsKey(arr[0])) {
+				source = originalNetworkUndir.addNode();
+				originalNetworkUndir.getRow(source).set(CyNetwork.NAME, arr[0]);
+				originalNetworkMap.put(arr[0], source.getSUID());
+			} else source = originalNetworkUndir.getNode(originalNetworkMap.get(arr[0]));
+
+			//to avoid adding same target
+			if (!originalNetworkMap.containsKey(arr[1])) {
+				target = originalNetworkUndir.addNode();
+				originalNetworkUndir.getRow(target).set(CyNetwork.NAME, arr[1]);
+				originalNetworkMap.put(arr[1], target.getSUID());
+			} else target = originalNetworkUndir.getNode(originalNetworkMap.get(arr[1]));
+
+			CyEdge edge = originalNetworkUndir.addEdge(source, target, true);
+			originalNetworkUndir.getRow(edge).set("edge_weight", Double.parseDouble(arr[2]));
+		}
+
+		//================================= setting up mixed network =====================================
+		originalNetworkMap = new HashMap<String, Long>();
+		originalNetworkMixed = support.getNetwork();
+		originalNetworkMixed.getDefaultEdgeTable().createColumn("edge_weight", Double.class, false);
+
+		//access input file
+		url = PathLinkerModelTest.class.getResource("/input/graph-mixed_human-interactome.txt");
+		file = new File(url.getFile());
+
+		//construct network based on input file
+		readerInput = new BufferedReader(new FileReader(file));
+
+		for (String line = readerInput.readLine(); line != null; line = readerInput.readLine()) {
+			String[] arr = line.split("\\s+");
+			CyNode source, target;
+
+			//to avoid adding same source
+			if (!originalNetworkMap.containsKey(arr[0])) {
+				source = originalNetworkMixed.addNode();
+				originalNetworkMixed.getRow(source).set(CyNetwork.NAME, arr[0]);
+				originalNetworkMap.put(arr[0], source.getSUID());
+			} else source = originalNetworkMixed.getNode(originalNetworkMap.get(arr[0]));
+
+			//to avoid adding same target
+			if (!originalNetworkMap.containsKey(arr[1])) {
+				target = originalNetworkMixed.addNode();
+				originalNetworkMixed.getRow(target).set(CyNetwork.NAME, arr[1]);
+				originalNetworkMap.put(arr[1], target.getSUID());
+			} else target = originalNetworkMixed.getNode(originalNetworkMap.get(arr[1]));
+
+			CyEdge edge = originalNetworkMixed.addEdge(source, target, true);
```

Comment:
This is not a "mixed" network, this is still just a directed network. You need to add edges as either directed or undirected according to the 4th column of the file.